google-account-plugin/src/com/google/cloud/tools/intellij/stats/GoogleUsageTracker.java

```diff
@@ -86,6 +112,18 @@ public GoogleUsageTracker() {
     AccountPluginInfoService pluginInfo = ServiceManager.getService(AccountPluginInfoService.class);
     externalPluginName = pluginInfo.getExternalPluginName();
     userAgent = pluginInfo.getUserAgent();
+    String intellijPlatformName = PlatformUtils.getPlatformPrefix();
+    String intellijPlatformVersion = ApplicationInfo.getInstance().getStrictVersion();
+    String cloudToolsPluginVersion = pluginInfo.getPluginVersion();
+    Map<String, String> systemMetaDataMap =
+        ImmutableMap.of(
+            PLATFORM_NAME_KEY, METADATA_ESCAPER.escape(intellijPlatformName),
+            PLATFORM_VERSION_KEY, METADATA_ESCAPER.escape(intellijPlatformVersion),
+            JDK_VERSION_KEY, METADATA_ESCAPER.escape(JDK_VERSION_VALUE),
+            OPERATING_SYSTEM_KEY, METADATA_ESCAPER.escape(OPERATING_SYSTEM_VALUE),
+            PLUGIN_VERSION_KEY, METADATA_ESCAPER.escape(cloudToolsPluginVersion));
+
+    metadataStringBuilder = METADATA_JOINER.appendTo(metadataStringBuilder, systemMetaDataMap);
```

Comment:
join returns a string. If you want to continue appending to a list of key value pairs, this is the only API I found.